chain/store/index.go

```diff
@@ -36,17 +35,14 @@ type ChainIndex struct {
 type loadTipSetFunc func(context.Context, types.TipSetKey) (*types.TipSet, error)
 
 func NewChainIndex(lts loadTipSetFunc) *ChainIndex {
-	sc, _ := lru.NewARC(DefaultChainIndexCacheSize)
 	return &ChainIndex{
-		skipCache:  sc,
+		indexCache: make(map[types.TipSetKey]*lbEntry),
```

Comment:
It's slower because you're not initializing the map to DefaultChainIndexCacheSize.

Reply:
Also note: I _think_ you can just use `lbEntry` by value to save some allocations which should speed things up even more.